ui/src/app/archived-workflows/components/archived-workflow-list/archived-workflow-list.tsx

```diff
@@ -181,8 +181,25 @@ export class ArchivedWorkflowList extends BasePage<RouteComponentProps<any>, Sta
         return params;
     }
 
+    private filterOutStateObject(state: State, blacklistedStates: string[]) {
+        const filteredState: State = Object.assign({}, state);
+        let key: keyof typeof state;
+
+        for (key in state) {
+            if (blacklistedStates.includes(key)) {
+                delete filteredState[key];
+            }
+        }
+
+        return filteredState;
+    }
+
+    private fetchBrowserStorageStateObject(state: State) {
+        return this.filterOutStateObject(state, ['workflows']);
+    }
+
     private saveHistory() {
-        this.storage.setItem('options', this.state, {} as State);
+        this.storage.setItem('options', this.fetchBrowserStorageStateObject(this.state), {} as State);
```

Comment:
I don't think this is the correct approach. A new objects should be constructed with only allowed fields.

Reply:
```typescript
interface Options {
        namespace: string
        name: string
        namePrefix: string
        selectedPhases: string[]
        selectedLabels: string[]
        minStartedAt: Date
        maxStartedAt: Date
        pagination: Pagination
}

Reply:
@alexec - But that'd mean we have to set the attributes manually for the new `options` variable, which is an `Options` interface object right? @juliev0 had raised a concern earlier that manually specifying attributes may have a maintenance overhead (if/when we add a new attribute to `State` interface, we'd also have to add it to the `Options` interface and in the function that converts the `State` object to `Options` object) - due to which I tried to automate the logic by taking `blacklistedStates` as an input.

Please lmk if `Options` is the preferred approach - I can update my PR with the same changes 😄

Reply:
Ping @alexec @sarabala1979 ^

Reply:
I think it preferable to construct new object to write lots of code to black list fields. Less code to maintain I think.